packages/select/src/components/query-list/queryList.tsx

```diff
@@ -79,15 +54,25 @@ export interface IQueryListRendererProps<T> extends IProps {
      */
     handleKeyUp: React.KeyboardEventHandler<HTMLElement>;
 
+    /**
+     * Change handler for query string. Attach this to an input element to allow
+     * `QueryList` to control the query.
+     */
+    handleQueryChange: React.ChangeEventHandler<HTMLInputElement>;
+
     /** Rendered elements returned from `itemListRenderer` prop. */
     itemList: React.ReactNode;
-
-    /** The current query string. */
-    query: string;
 }
 
 export interface IQueryListState<T> {
+    /** The currently focused item (for keyboard interactions). */
+    activeItem: T | undefined;
+
+    /** The original `items` array filtered by `itemListPredicate` or `itemPredicate`. */
     filteredItems: T[];
+
+    /** The current query string. */
+    query: string;
```

Comment:
how is this resolved with query from props? is this just to allow control/uncontrolled variants?